### ISSUE TITLE
j.modular - namespace/build : discovery is sent 3 times

### ISSUE THREAD

**Issue by avilleret** [jamoma/JamomaMax#963]
when I send `[namespace/build(` message, `j.modular` monitor shows that discovery is sent 3 times successively even if I receive an answer for the three.

Why is it doing so ?
If this is to be sure that the request is well received, I think such a repeat should be handle by the user or at least we could choose how many repeat we need and the amount of time to wait for an answer

but maybe I'm wrong and this repetition do have a good reason to be there ?

**Comment by theod** [comment]
yes it is hard coded into the Minuit protocol but it shouldn't try again if it receive the answer. or maybe the answer arrived too late so it tries again. I agree this should be an attribute of the protocol.

**Comment by avilleret** [comment]
I noticed that not only discovery but all commands seem to be send three times

**Comment by theod** [comment]
this still sounds like if network messages are too slow and so answer arrived too late (after 1s) and so another query is sent.

**Comment by theod** [comment]
in order to be sure this trouble is due to the default time out, could you change it into MinuitIncludes.h
there is a #define DEFAULT_TIMEOUT 50. if you try 1000 it should fix your problem and then I'll start to create a protocol attribute for this.
